src/server/routes/emailTemplates.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Router } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import { render } from 'mjml-react';
+
+import { ResetPassword } from '../../email/templates/ResetPassword/ResetPassword';
+import { ResetPasswordText } from '../../email/templates/ResetPassword/ResetPasswordText';
+import { Verify } from '../../email/templates/Verify/Verify';
+import { VerifyText } from '../../email/templates/Verify/VerifyText';
+import { AccidentalEmail } from '../../email/templates/AccidentalEmail/AccidentalEmail';
+import { AccidentalEmailText } from '../../email/templates/AccidentalEmail/AccidentalEmailText';
+
+const router = Router();
+
+const emailTemplateTypes = [
+  'reset-password',
+  'verify',
+  'accidental-email',
+] as const;
+type EmailTemplateType = typeof emailTemplateTypes[number];
+
+type EmailRenderResult = {
+  plain: string;
+  html: string;
+};
+
+const renderEmailTemplate = (
+  template: EmailTemplateType,
+): EmailRenderResult | undefined => {
+  switch (template) {
+    case 'reset-password':
+      return {
+        plain: ResetPasswordText(),
+        html: render(ResetPassword()).html,
```

Comment:
rather than rendering the HTML on each call to the route, an optimisation would be render the HTML earlier and return the html string. Calling the render method on each request is expensive, and a route to a possible DoS attack.

The HTML is static too, so doesn't need to be rendered over and over again. 

I suggest either:
a) render the html at app startup and return the HTML string for that email
b) memoize the output of the render method

Reply:
This has now been done by rendering the HTML at app startup!